From 0b8b0aee91f1dd172620fad167c6776b49946876 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Thu, 26 Feb 2015 11:58:33 +0100
Subject: [LIBREPORT PATCH] problem_data: add type to problem_data in add_basic

satyr-0.16 requires the file named type in dump directories, this might not be
necessary but I want to be sure that type is always created.

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/lib/problem_data.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lib/problem_data.c b/src/lib/problem_data.c
index 4a45b6c..e80569a 100644
--- a/src/lib/problem_data.c
+++ b/src/lib/problem_data.c
@@ -60,12 +60,15 @@ problem_data_t *new_problem_data(void)
 void add_basics_to_problem_data(problem_data_t *pd)
 {
     const char *analyzer = get_problem_item_content_or_NULL(pd, FILENAME_ANALYZER);
+    const char *type = get_problem_item_content_or_NULL(pd, FILENAME_TYPE);
     if (analyzer == NULL)
     {
         analyzer = "libreport";
         add_to_problem_data(pd, FILENAME_ANALYZER, analyzer);
     }
-    add_to_problem_data(pd, FILENAME_TYPE, analyzer);
+
+    if (type == NULL)
+        add_to_problem_data(pd, FILENAME_TYPE, analyzer);
 
     /* If application didn't provide dupe hash, we generate it
      * from all components, so we at least eliminate the exact same
-- 
1.8.3.1

